Relax the bound to make 9.8 compatible. The requirement of the new version
is irrelevant as the problem doesn't affect OpenBSD:
https://haskell.github.io/security-advisories/advisory/HSEC-2024-0003.html

Index: hledger.cabal
--- hledger.cabal.orig
+++ hledger.cabal
@@ -159,7 +159,7 @@ library
     , megaparsec >=7.0.0 && <9.7
     , microlens >=0.4
     , mtl >=2.2.1
-    , process >=1.6.19.0
+    , process >=1.6.18.0
     , regex-tdfa
     , safe >=0.3.20
     , shakespeare >=2.0.2.2
@@ -209,7 +209,7 @@ executable hledger
     , megaparsec >=7.0.0 && <9.7
     , microlens >=0.4
     , mtl >=2.2.1
-    , process >=1.6.19.0
+    , process >=1.6.18.0
     , regex-tdfa
     , safe >=0.3.20
     , shakespeare >=2.0.2.2
@@ -260,7 +260,7 @@ test-suite unittest
     , megaparsec >=7.0.0 && <9.7
     , microlens >=0.4
     , mtl >=2.2.1
-    , process >=1.6.19.0
+    , process >=1.6.18.0
     , regex-tdfa
     , safe >=0.3.20
     , shakespeare >=2.0.2.2
@@ -310,7 +310,7 @@ benchmark bench
     , megaparsec >=7.0.0 && <9.7
     , microlens >=0.4
     , mtl >=2.2.1
-    , process >=1.6.19.0
+    , process >=1.6.18.0
     , regex-tdfa
     , safe >=0.3.20
     , shakespeare >=2.0.2.2
